containers/camera.js

```diff
@@ -50,7 +50,7 @@ class CameraComponent extends Component {
   }
 
   componentWillReceiveProps (nextProps) {
-    if (this.props.isRecording !== nextProps.isRecording && nextProps.token !== '') {
+    if (nextProps.isRecording === false && nextProps.recordedFile !== '') {
```

Comment:
No changes requested: actually you can check if it has changed by doing `this.props.isRecording !== nextProps.isRecording`.